+neuron/Vector.m

```diff
@@ -28,6 +28,28 @@
 
         end
 
+        % See https://nl.mathworks.com/help/matlab/matlab_oop/code-patterns-for-subsref-and-subsasgn-methods.html
+        function self = subsasgn(self, S, varargin)
+            % Assigning a Vector element by index.
+            if (length(S) == 1 && S(1).type == "()")
+                element_id = S(1).subs{:};
+                self.call_method_hoc("set", "Object", element_id-1, varargin{:});
+            % Are we trying to directly access a class property?
+            elseif (isa(S(1).subs, "char") && length(S) == 1 && isprop(self, S(1).subs))
+                self.(S(1).subs) = varargin{:};
+            end
+        end
+
+        function varargout = subsref(self, S)
+            % Are we trying to access a Vector element?
```

Comment:
To discuss with @ramcdougal / check what we can do here /put in an issue for now: currently it is implemented that Vector data is accessed through vec_ref(1). What if we create an array of Vectors, how will that arrive here in subsref?

Or is the length(S) == 1 check already enough to cover this correctly?

```
vec_1 = n.Vector(...)
vec_2 = n.Vector(...)
vecs = [vec_1, vec_2];
% then when we do
vecs(1)
% what do we get? We probably want to access the data directly as well, so with
vecs(1)(5)
```

Reply:
Created #55